web/src/app/app.component.ts

```diff
@@ -0,0 +1,160 @@
+declare let require: any;
+declare var web3: any;
+import { Component } from '@angular/core';
+const etherlime = require('etherlime');
+const ethers = require('ethers');
+const ToDo = require('../../../build/ToDoManager.json');
+
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+
+
+export class AppComponent {
+  title = 'My Decentralized App';
+
+  public successMessage: string;
+  public infoMessage: string;
+  public errorMessage: string;
+  public contractInstance: any;
+  public contractAddress = '0xc9707E1e496C12f1Fa83AFbbA8735DA697cdBf64';
+  public toDoStatus: any;
+  public inProgressStatus: any;
+  public doneStatus: any;
+  public inactiveButton;
+
+  constructor() {
+    this.toDoStatus = [];
+    this.inProgressStatus = [];
+    this.doneStatus = [];
+
+    etherlime.ContractAt(ToDo, this.contractAddress).then((result) => {
+      this.contractInstance = result;
+      this.successMessage = 'The contract has been set and is ready to interact with it!';
+      this.getToDoStatuses();
+    }).catch((e) => {
+      this.errorMessage = e.message;
+    });
+
+  }
+
+  private async getToDoIndex(_toDo) {
+    const indexCounter = await this.contractInstance.indexCounter();
+    for (let i = 0; i < indexCounter.toNumber(); i++) {
```

Comment:
Having function being called in the condition of a for actually forces the for to execute this function multiple times. It is better to take the indexCounter value into a variable and put it in the for instead of constantly calling toNumber()